### ISSUE TITLE
Illegal reflective access when running jOOQ on Java 10

### ISSUE THREAD

**Issue by MirkoTeran** [jOOQ/jOOQ#7594]
### Expected behavior and actual behavior:
Running any kind of app using jOOQ 3.11.0 compiled with java 10 gives the following warning:
> WARNING: An illegal reflective access operation has occurred
> WARNING: Illegal reflective access by org.jooq.tools.reflect.Reflect (file:/C:/Users/teranm/.m2/repository/org/jooq/jooq/3.11.0/jooq-3.11.0.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class)
> WARNING: Please consider reporting this to the maintainers of org.jooq.tools.reflect.Reflect
> WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
> WARNING: All illegal access operations will be denied in a future release

Compiling same file with java 8 does not generate this warning if running with java 8 or java 10. Warning also doesn't appear when using older version of jOOQ (tested with 3.10.7).

Similar warning also appears when trying to generate classes, but that is already noted in issue #7579 

### Steps to reproduce the problem (include example code if possible):
Simplest application to reproduce the problem:
```java
public static void main(String[] args) {
    System.out.println(
        DSL.using("jdbc:h2:~/playground", "sa", "sa")
        .select(DSL.val(9001).as("over"))
        .fetch()
    );
}
```

Only 2 dependencies in maven are jOOQ and H2.

Compile and run with jOOQ 3.11.0 and java 10.

### Versions:

- jOOQ: 3.11.0
- Java: 10 (tested with orace and zulu builds)
- Database (include vendor): H2
- OS: Windows10
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks for reporting. jOOQ 3.12 will cross release a modular JDK 9+ version along with legacy JDK 6 and 8 versions. The JDK 9+ version will not execute the logic that produces the warning. In fact, this warning is already irrelevant in the current version of jOOQ as any possible exception will be intercepted and handled, such that the JDK 8 version of jOOQ can also run on JDK 9+:
https://github.com/jOOQ/jOOQ/blob/version-3.11.1/jOOQ/src/main/java/org/jooq/tools/reflect/Reflect.java#L184

However, I agree the warning isn't pretty. Perhaps we can add an additional runtime check to avoid it entirely

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. Backport for 3.11.2 scheduled: #7601.

**Comment by lukaseder** [comment]
The suggested fix causes problems when calling defaulted methods on JDK 9. Better live with the warning for now. A thorough fix is to include JDK 9 specific logic in a proper JDK 9 enabled build. This will be fixed for jOOQ 3.12 only, though: #7602
